internal/services/fingerprint/consumer.go

```diff
@@ -74,8 +88,19 @@ func RunConsumer(ctx context.Context, settings *config.Settings, logger *zerolog
 
 	go func() {
 		for {
-			if err := group.Consume(ctx, []string{settings.DeviceFingerprintTopic}, c); err != nil {
-				logger.Warn().Err(err).Msg("Consumer group session ended.")
+			if err := deviceGroup.Consume(ctx, []string{settings.DeviceFingerprintTopic}, c); err != nil {
+				logger.Warn().Err(err).Msgf("Consumer group session ended: %s", settings.DeviceFingerprintTopic)
+			}
+			if ctx.Err() != nil {
+				return
+			}
+		}
+	}()
+
+	go func() {
```

Comment:
Could we try using this? Just spawn two?

https://github.com/DIMO-Network/shared/blob/01c0fa977163f1169266e4521656cb8ec5f2e81c/kafka/group.go#L42

Getting rid of the `switch` above seems good to me.